papers/source/d1664.bs

```diff
@@ -243,7 +257,22 @@ Append to §17.3.1 General [[support.limits.general](http://eel.is/c++draft/supp
 Insert into §24.4.4 Ranges [[range.range](http://eel.is/c++draft/range.req#range.range)]'s after clause 7, one additional clause:
 
 <blockquote>
-<ins><sup>8</sup> The <i>pair-reconstructible-range</i> and <i>range-reconstructible-range</i> concepts denote ranges whose iterator and sentinel pair can be used to construct an object of the type from which they originated. That is, given an expression `E` such that `decltype((E))` is `T`, then `T(ranges::begin(E), ranges::end(E))` and `T(ranges::subrange(ranges::begin(E), ranges::end(E)))` are well-formed, respectively.</ins>
+<ins><sup>8</sup> The exposition-only <i>pair-reconstructible-range</i> and <i>range-reconstructible-range</i> concepts denote ranges whose iterator and sentinel pair can be used to efficiently construct an object of the type from which they originated.
+
+```
+template <typename R>
+concept pair-reconstructible-range =
+    Range<R> &&
+    forwarding-range<std::remove_reference_t<R>> &&
+    std::Constructible<R, iterator_t<R>, sentinel_t<R>>;
+
+template <typename R>
+concept range-reconstructible-range =
+    Range<R> &&
+    forwarding-range<std::remove_reference_t<R>> &&
+    std::Constructible<R, std::ranges::subrange<iterator_t<R>, sentinel_t<R>>>;
+```
+</ins>

```

Comment:
I realised the wording changes didn't actually include the wording proposed above so I added it here, too.

Reply:
Thanks for catching that!